pandas/core/frame.py

```diff
@@ -4478,19 +4478,20 @@ def _maybe_casted_values(index, labels=None):
 
     @Appender(_shared_docs["isna"] % _shared_doc_kwargs)
     def isna(self) -> "DataFrame":
-        return super().isna()
+        result = self._constructor(self._data.isna(func=isna))
```

Comment:
This should be isna or isna_old depending on the settings?

Reply:
(and since that is handled in the `isna` function, I am not fully sure it will be easier to do it here)

Reply:
> This should be isna or isna_old depending on the settings?

No, the _isna/_isna_old is reflected in `isna`

Reply:
Ah, I missed that. So it's basically calling that recursively on its parts. OK!